src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationClientFactory.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Azure.Core;
+using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Azure;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal class AzureAppConfigurationClientFactory : IAzureClientFactory<ConfigurationClient>
+    {
+        private readonly ConfigurationClientOptions _clientOptions;
+
+        private readonly TokenCredential _credential;
+        private readonly IEnumerable<string> _connectionStrings;
+
+        public AzureAppConfigurationClientFactory(
+            IEnumerable<string> connectionStrings,
+            ConfigurationClientOptions clientOptions)
+        {
+            if (connectionStrings == null || !connectionStrings.Any())
+            {
+                throw new ArgumentException(nameof(connectionStrings));
+            }
+
+            _connectionStrings = connectionStrings;
+
+            _clientOptions = clientOptions ?? throw new ArgumentNullException(nameof(clientOptions));
+        }
+
+        public AzureAppConfigurationClientFactory(
+            TokenCredential credential,
+            ConfigurationClientOptions clientOptions)
+        {
+            _credential = credential ?? throw new ArgumentNullException(nameof(credential));
+            _clientOptions = clientOptions ?? throw new ArgumentNullException(nameof(clientOptions));
+        }
+
+        public ConfigurationClient CreateClient(string endpoint)
+        {
+            if (string.IsNullOrEmpty(endpoint))
+            {
+                throw new ArgumentNullException(nameof(endpoint));
+            }
+
+            if (!Uri.TryCreate(endpoint, UriKind.Absolute, out Uri uriResult) || !(uriResult.Scheme == Uri.UriSchemeHttp || uriResult.Scheme == Uri.UriSchemeHttps))
```

Comment:
Why do we validate scheme?

Reply:
Our store endpoints have the `https` scheme, so I thought it should be validated

Reply:
done, removed it